src/typing/env_js.ml

```diff
@@ -116,9 +146,12 @@ let pop_env () =
   env := List.tl !env
 
 let flat_env () =
```

Comment:
This is the only change I'm not sure I agree with. `set_in_env` and `set_in_scope` actually mutate the relevant refs, but `flat_env` returns a new value and leaves the env and global scope unchanged. `flattened_env` might be preferable to `flat_env`, but `flatten_env` makes it seem like this method would change env when defined adjacent to other mutating methods.